### ISSUE TITLE
Train performance is poor？

### ISSUE THREAD

**Issue by wuwo007** [SizheQiu/Seq2Topt#4]
Dear Authors，

Thank you very much for sharing such excellent work. 

I am trying to retrain the seq2Topt model. Initially, I used the training code provided by the author, Seq2Topt-main/code/run_train.py, with the training and test datasets located at Seq2Topt-main/data/Topt/train.csv and Seq2Topt-main/data/Topt/test.csv, respectively. 
However, my training results differ significantly from those shared by the author.
My training results is as follows:
        We use CUDA!
        for win_size =3
        epoch: 0/30;  rmse test: 0.124588; r2 test: 0.355659
        epoch: 5/30;  rmse test: 0.113278; r2 test: 0.467338
        epoch: 10/30;  rmse test: 0.112215; r2 test: 0.477286
        epoch: 15/30;  rmse test: 0.110522; r2 test: 0.492941
        epoch: 20/30;  rmse test: 0.109616; r2 test: 0.501215
        epoch: 25/30;  rmse test: 0.113065; r2 test: 0.46934
        Done.
        for win_size = 5
        epoch: 0/30;  rmse test: 0.126435; r2 test: 0.33642
        epoch: 5/30;  rmse test: 0.13823; r2 test: 0.206829
        epoch: 10/30;  rmse test: 0.110853; r2 test: 0.489898
        epoch: 15/30;  rmse test: 0.125202; r2 test: 0.349293
        epoch: 20/30;  rmse test: 0.110481; r2 test: 0.493319
        epoch: 25/30;  rmse test: 0.114612; r2 test: 0.454714
        Done.
        for win_size = 7
        epoch: 0/30;  rmse test: 0.198667; r2 test: -0.638379
        epoch: 5/30;  rmse test: 0.114635; r2 test: 0.454497
        epoch: 10/30;  rmse test: 0.116315; r2 test: 0.438394
        epoch: 15/30;  rmse test: 0.116269; r2 test: 0.438834
        epoch: 20/30;  rmse test: 0.119224; r2 test: 0.409948
        epoch: 25/30;  rmse test: 0.109374; r2 test: 0.503422
        Done.

Could you please advise on what additional details I should pay attention to?

Thanks.

**Comment by SizheQiu** [comment]
Did you use the same random seed ? 
The version of pytorch is also important, the one used in this project was PyTorch/1.12.0-foss-2022a-CUDA-11.7.0

**Comment by SizheQiu** [comment]
The version of pytorch is also important, the one used in this project was PyTorch/1.12.0-foss-2022a-CUDA-11.7.0

**Comment by wuwo007** [comment]
I used the random seed in the script "Seq2Topt-main/code/run_train.py". 
I do not think the pytorch version will affect the performance. Did you find the pytorch version affect the performance?

**Comment by SizheQiu** [comment]
Yes, different versions of PyTorch can significantly affect performance. This is due to factors like changes in the underlying CUDA toolkit, libraries, and PyTorch's internal implementation itself.
